build.sbt

```diff
@@ -154,3 +155,19 @@ lazy val microsite = project.module
       "white-color"     -> "#FFFFFF"
     )
   )
+
+lazy val commitSha = Process("git rev-parse --short HEAD").lineStream.head
+
+releaseVersion := ((version: String) => s"${version}-${commitSha}" )
+releaseTagName := s"v${version.value}"
+
+releaseProcess := Seq[ReleaseStep](
+  checkSnapshotDependencies,
+  inquireVersions,
+  runClean,
+  runTest,
+  setReleaseVersion,
```

Comment:
We don't want these: `setReleaseVersion`, `tagRelease`, `pushChanges`

Reply:
- sbt-release builds desired artifact based on new version set on `version.sbt`, which is set by `setReleaseVersion`. So, we want `setReleaseVersion`. but I should look into reverting the `version.sbt` to original version once build is successful.

- `tagRelease` is used to create a tag for each build, but depends what responsibility we want with `sbt` and what we want manual. I had a comment to do the same [using bash script](https://github.com/scalaz/scalaz-zio/issues/145#issuecomment-408970343) as I can create tag from scripting as well. 

I would be fine with `sbt-release` creating tags, unless contributors have permissions to push their dev tags (with `git push`) which will cause problems.

- I agree we do not need `pushChanges` and don't want to keep pushing dev tags. which means we need to either introduce some sort of flag which will let push the tags from CI/CD.
or go write scrip and simply `git push --tags`